controllers/rclone_mover_test.go

```diff
@@ -0,0 +1,671 @@
+								Eventually(func() bool {
+									Expect(k8sClient.Get(ctx, utils.NameFor(rs), rs)).To(Succeed())
+									if rs.Status == nil || rs.Status.NextSyncTime.IsZero() {
+										return false
+									}
+									return true
+								}, maxWait, interval).Should(BeTrue())
+							})
+						})
+					})
+
+					It("Uses the Source PVC as the sync source", func() {
+						Eventually(func() error {
+							return k8sClient.Get(ctx, utils.NameFor(job), job)
+						}, maxWait, interval).Should(Succeed())
+						// look for the source PVC in the job's volume list
+						volumes := job.Spec.Template.Spec.Volumes
+						found := false
+						for _, v := range volumes {
+							if v.PersistentVolumeClaim != nil && v.PersistentVolumeClaim.ClaimName == srcPVC.Name {
+								found = true
+							}
+						}
+						Expect(found).To(BeTrue())
+						Expect(srcPVC).NotTo(beOwnedBy(rs))
+					})
+				})
+
+				When("copyMethod of Clone is specified", func() {
+					BeforeEach(func() {
+						rs.Spec.Rclone.ReplicationSourceVolumeOptions.CopyMethod = scribev1alpha1.CopyMethodClone
+					})
+
+					// attempts to invoke rloneReconcile
+					//nolint:dupl
+					It("creates a clone of the source PVC as the sync source", func() {
+						Eventually(func() error {
+							return k8sClient.Get(ctx, utils.NameFor(job), job)
+						}, maxWait, interval).Should(Succeed())
+						volumes := job.Spec.Template.Spec.Volumes
+						pvc := &corev1.PersistentVolumeClaim{}
+						pvc.Namespace = rs.Namespace
+						found := false
+						for _, v := range volumes {
+							if v.PersistentVolumeClaim != nil {
+								found = true
+								pvc.Name = v.PersistentVolumeClaim.ClaimName
+							}
+						}
+						Expect(found).To(BeTrue())
+						Expect(k8sClient.Get(ctx, utils.NameFor(pvc), pvc)).To(Succeed())
+						Expect(pvc.Spec.DataSource.Name).To(Equal(srcPVC.Name))
+						Expect(pvc).To(beOwnedBy(rs))
+					})
+
+					Context("Pausing an rclone sync job", func() {
+						parallelism := int32(0)
+						BeforeEach(func() {
+							rs.Spec.Paused = true
+						})
+						It("job will be created but won't start", func() {
+							Eventually(func() error {
+								return k8sClient.Get(ctx, utils.NameFor(job), job)
+							}, maxWait, interval).Should(Succeed())
+							Expect(*job.Spec.Parallelism).To(Equal(parallelism))
+						})
+					})
+				})
+			})
+		})
+		When("rclone is given an incorrect config", func() {
+			var emptyString = ""
+			BeforeEach(func() {
+				rs.Spec.Rclone = &scribev1alpha1.ReplicationSourceRcloneSpec{
+					ReplicationSourceVolumeOptions: scribev1alpha1.ReplicationSourceVolumeOptions{
+						CopyMethod: scribev1alpha1.CopyMethodNone,
+					},
+					RcloneConfig:        &emptyString,
+					RcloneConfigSection: &emptyString,
+					RcloneDestPath:      &emptyString,
+				}
+			})
+			When("All fields are empty", func() {
+				It("should not start", func() {
+					Expect(k8sClient.Get(ctx, utils.NameFor(job), job)).NotTo(Succeed())
```

Comment:
Need a Consistently() here since it may take a while for the controller to start the job
(and the other tests below, too)